### ISSUE TITLE
Support for th-abstraction-0.3.1.0?

### ISSUE THREAD

**Issue by peti** [monadfix/microlens#112]
Hi, 

would it be possible to get a minor update for `microlens-th` that supports `th-abstraction-0.3.1.0`?

Best regards,
Peter

**Comment by neongreen** [comment]
I have published a new revision on Hackage.

**Comment by peti** [comment]
Wow, that was quick! Thank you very much.

**Comment by kquick** [comment]
This was done as a revision and didn't affect the release number, so I get intermittent build failures depending on whether the system+directory package cache has the old 0.4.2.3 version or the new 0.4.2.3 version.  Would it be possible to get a release with an actual version change to reliably ensure builds succeed?

**Comment by neongreen** [comment]
@kquick Sure, I will do a patch release. I'm curious, though – how does your build process look like? What tools are you using? (Perhaps I would need to adjust the release process for all my packages, not just microlens.)

**Comment by kquick** [comment]
Sorry for the delay @neongreen, I was on vacation.  I encountered this in two ways.  The first was a local cabal new-build that had cached the old version but then the need for th-abstraction 0.3.x was driven by other needs, but it failed on microlens-th.  After I deleted the dist-newstyle directory it re-downloaded everything and got the new version.

The other problem is building with nix.  The nix store may already have the old image, since there's no new version to direct it to fetch the new image, and it believes that the build should be possible but it fails due to the new th-abstraction that isn't allowed by the old image.  My workaround is to add a patch to the nix build that will bump the th-abstraction upper bound if it's low, and no-op if it's the newer version.
